exporters/sender/okhttp/src/test/java/io/opentelemetry/exporter/sender/okhttp/internal/OkHttpSenderTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.sender.okhttp.internal;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.opentelemetry.exporter.internal.InstrumentationUtil;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+abstract class OkHttpSenderTest<T> {
+
+  @BeforeEach
+  void setUp() {
+    OkHttpUtil.propagateContextInDispatcher = true;
```

Comment:
Is this only needed in tests or is it needed all the time?

Reply:
Yes, it's only needed for tests. I know it's a bit ugly but it's the least intrusive way I found to pass the Context to the thread used by okhttp for these tests. Still, I'm not fully happy with it so I'm open to alternatives.